internal/controller/isbservicerollout/isbservicerollout_controller.go

```diff
@@ -256,22 +262,16 @@ func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceR
 			numaLogger.WithValues("recyclable isbservices", recyclableISBSvcs).Debug("can't create 'promoted' isbservice yet; need to wait for recyclable isbservices to be deleted")
 			requeueDelay = common.DefaultRequeueDelay
 		} else {
-
 			// create an object as it doesn't exist
-			newISBServiceDef, err := r.makeTargetISBServiceDef(ctx, isbServiceRollout)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error generating ISBService: %v", err)
-			}
-
 			numaLogger.Debugf("ISBService %s/%s doesn't exist so creating", isbServiceRollout.Namespace, newISBServiceDef.GetName())
 			isbServiceRollout.Status.MarkPending()
 			if newISBServiceDef != nil {
-
 				if err = r.createPromotedISBService(ctx, isbServiceRollout, newISBServiceDef); err != nil {
 					return ctrl.Result{}, err
 				}
 
 				isbServiceRollout.Status.MarkDeployed(isbServiceRollout.Generation)
+				r.customMetrics.IncISBSvcProgressiveResults(newISBServiceDef.GetName(), isbServiceRollout, "false")
```

Comment:
I don't think this is right. Here you've just created a new "promoted" isbsvc and in the function, you are looking at the "Upgrading" isbsvc. In fact, this metric only applies to "upgrading" children (up until they are first promoted). So, you should never see "my-isbsvc-0" for this metric.